apache2/apache2_util.c

```diff
@@ -461,7 +462,25 @@ static void internal_log_ex(request_rec *r, directory_config *dcfg, modsec_rec *
         else requestheaderhostname = "";
 
 #ifdef WAF_JSON_LOGGING_ENABLE
-        send_waf_log(msr->modsecurity->wafjsonlog_lock, dcfg->wafjsonlog_fd, str1, r->useragent_ip ? r->useragent_ip : r->connection->client_ip, log_escape(msr->mp, r->uri), dcfg->is_enabled, (char*)msr->hostname, unique_id, r);
+        if (msc_waf_log_reopened){
+            rc = apr_file_open(&msc_waf_log_fd, msc_waf_log_path,
+                   APR_WRITE | APR_APPEND | APR_CREATE | APR_BINARY,
+                   CREATEMODE | APR_WREAD, msc_waf_log_cmd->pool);
+
+            if (rc != APR_SUCCESS) {
+                #if AP_SERVER_MAJORVERSION_NUMBER > 1 && AP_SERVER_MINORVERSION_NUMBER > 2
+                	ap_log_rerror(APLOG_MARK, APLOG_ERR | APLOG_NOERRNO, 0, r,
+                            "MoSecurity not able to reopen %s file", WAF_LOG_UTIL_FILE);
+                #else
+                        ap_log_error(APLOG_MARK, APLOG_ERR | APLOG_NOERRNO, 0, r->server,
```

Comment:
Since we are using modern Apache, can we just abandon this entire #else branch altogether?

Reply:
I think we should follow the same convention as line 490, so people don't use moder Apache could also use it